packages/core/src/prompts/lesson-assistant/index.ts

```diff
@@ -57,20 +58,21 @@ export const getPromptParts = (props: TemplateProps): TemplatePart[] => {
 
   const parts: (TemplatePart | undefined)[] = [
     context,
-    currentLessonPlan,
     task,
-    body,
-    props.useRag ? rag : undefined,
-    props.baseLessonPlan ? basedOn : undefined,
     props.responseMode === "interactive" ? interactingWithTheUser : undefined,
     props.responseMode === "interactive" ? lessonComplete : undefined,
     props.responseMode === "interactive"
       ? endingTheInteractionSection
       : undefined,
+    body,
+    currentLessonPlan,
+    props.useRag ? rag : undefined,
+    props.baseLessonPlan ? basedOn : undefined,
     americanToBritishSection,
     languageAndVoice,
     props.isUsingStructuredOutput ? undefined : schema,
     response,
+    props.responseMode === "interactive" ? promptingTheUser : undefined,
```

Comment:
This re-orders the prompt sections and breaks the response out, putting it last. This seems to fix a lot of the isues about the response text being long / incorrect